libs/geo/src/main/java/org/elasticsearch/geo/utils/BitUtil.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.geo.utils;
+
+/**
+ * Utilities for common Bit twiddling methods. Borrowed heavily from Lucene (org.apache.lucene.util.BitUtil).
+ */
+public class BitUtil {  // magic numbers for bit interleaving
```

Comment:
Just for my understanding, why not just use these specific methods directly from Lucene?

Reply:
Like `libs/core` (where `IOUtils` is largely a copy of `o.a.l.util.IOUtils`) the geo library doesn't have access to lucene classes. Honestly, I'm not sure if this is by design (maybe @rjernst knows?) but I didn't think these few utility methods warranted an internal library dependency on lucene. Maybe it does?

Reply:
Correct, libs/geo is independent and doesn't depend on Lucene.